src/session/session_api.c

```diff
@@ -1449,8 +1456,21 @@ __wt_session_range_truncate(
             goto done;
     }
 
+    /* Create a copy of the start and stop cursors for error-checking purposes. */
+#ifdef HAVE_DIAGNOSTIC
+    debug_start = debug_stop = NULL;
+    if (start != NULL)
+        WT_ERR(__session_open_cursor((WT_SESSION *)session, NULL, start, NULL, &debug_start));
+    if (stop != NULL)
+        WT_ERR(__session_open_cursor((WT_SESSION *)session, NULL, stop, NULL, &debug_stop));
+#endif
+
     WT_ERR(__wt_schema_range_truncate(session, start, stop));
 
+#ifdef HAVE_DIAGNOSTIC
+    is_truncate = true;
```

Comment:
Can this be set in the previous `#ifdef HAVE_DIAGNOSTIC` section? Before the call to `__wt_schema_range_truncate`.

Reply:
Yes good idea, I think maybe it should be set there instead, just before opening the debug cursors.